src/train.py

```diff
@@ -1,20 +1,66 @@
 from datasets import CasiaSurfDataset
 from torch.utils.data import DataLoader, SubsetRandomSampler
 import numpy as np
+import torch
+from torch import optim, nn
+from torchvision import models
+import argparse
 
 
-def get_loader(dataset, batch_size=64, split=(4, 1)):
+def get_loader(dataset, batch_size=8, split=(4, 1)):
     val_q = len(dataset) // sum(split) * split[1]
 
     val_indices = np.random.choice(list(range(len(dataset))), val_q)
     train_indices = list(set(range(len(dataset))).difference(val_indices))
-    sampler = SubsetRandomSampler(train_indices)
-    return DataLoader(dataset, sampler=sampler, batch_size=batch_size)
+    train_sampler = SubsetRandomSampler(train_indices)
+    vaild_sampler = SubsetRandomSampler(val_indices)
+    return DataLoader(dataset, sampler=train_sampler, batch_size=batch_size),\
+        DataLoader(dataset, sampler=vaild_sampler, batch_size=batch_size)
 
 
 if __name__ == '__main__':
-    dataset = CasiaSurfDataset()
-    data_loader = get_loader(dataset)
-    for images, labels in enumerate(data_loader):
-        print(images, labels)
-        raise
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument('--protocol', type=int, required=True)
+    argparser.add_argument('--epochs', type=int, default=10)
+    argparser.add_argument('--checkpoint', type=str)
+    argparser.add_argument('--batch_size', type=int, required=True)
+    argparser.add_argument('--eval_every', type=int, default=1)
+    args = argparser.parse_args()
+    dataset = CasiaSurfDataset(args.protocol)
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    train_loader, val_loader = get_loader(dataset)
+    model = models.mobilenet_v2(
+        num_classes=2) if not args.checkpoint else torch.load(args.checkpoint)
+    model = model.to(device)
+    optimizer = optim.Adam(model.parameters())
```

Comment:
learning rate should be a parameter to argparser

Reply:
I agree that we should give an ability to specify this somewhere outside. Isn't config file a better place for a model parameter?